app/lucy/src/app/services/export/export.service.ts

```diff
@@ -0,0 +1,103 @@
+import { Injectable } from '@angular/core';
+import { saveAs } from 'file-saver';
+import * as Papa from 'papaparse/papaparse.min.js';
+import { ApiService, APIRequestMethod } from '../api.service';
+import { AppConstants } from 'src/app/constants';
+
+export enum ExportType {
+  WatercraftRiskAssessment,
+  Shift
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ExportService {
+
+  constructor(private api: ApiService) { }
+
+  /**
+   * Download CSV export of data type specified.
+   */
+  public exportCSV(type: ExportType) {
+    this.exportCSVfrom(this.urlForExportType(type), this.fileNameForExportType(type));
+  }
+
+  public async getInspectAppExportData(type: ExportType): Promise<any> {
+    const data = await this.api.request(APIRequestMethod.GET, this.urlForExportType(type), null);
+    if (!data.success) {
+      console.log('Error: Couldnt fetch data');
+      return;
+    }
+    return data.response
+  }
+
+  /**
+   * Returns the endpoint for the given ExportType.
+   * @param type ExportType
+   */
+  private urlForExportType(type: ExportType): string {
+    switch (type) {
+      case ExportType.WatercraftRiskAssessment:
+        return AppConstants.API_WatercraftAssessment_Export;
+      case ExportType.Shift:
+        return AppConstants.API_Shift_Export;
+    }
+  }
+
+  /**
+   * Returns the endpoint for the given ExportType.
+   * @param type ExportType
+   */
+  private fileNameForExportType(type: ExportType): string {
+    switch (type) {
```

Comment:
It's not a big deal, but it could save some lines of the code. Instead of doing this as a `switch`, what you can do here is, 

`return ExportType.Shift ? shiftReportName : waterCraftName`

Although, the above approach is good only when you have two options. But when you have more than two, I'd recommend doing something like this

return {
   ExportType.Shift: Shiftname.bla-bla-bla,
   ExportType.WaterRisk: ....,
  ......
}[type]